core/chains/evm/client/rpc_client.go

```diff
@@ -115,6 +92,49 @@ func NewRPCClient(
 	return r
 }
 
+func (r *rpcClient) SubscribeToHeads(ctx context.Context) (<-chan *evmtypes.Head, commontypes.Subscription, error) {
+	channel := make(chan *evmtypes.Head)
+	sub, err := r.Subscribe(ctx, channel)
+	return channel, sub, err
+}
+
+func (r *rpcClient) SubscribeToFinalizedHeads(_ context.Context) (<-chan *evmtypes.Head, commontypes.Subscription, error) {
+	// Use Poller to get finalized heads
+	pollFunc := func(ctx context.Context) (*evmtypes.Head, error) {
+		return r.LatestFinalizedBlock(ctx)
+	}
+	interval := r.cfg.FinalizedBlockPollInterval()
+	timeout := interval
+	poller, channel := commonclient.NewPoller[*evmtypes.Head](interval, pollFunc, timeout, r.rpcLog)
```

Comment:
Hm, should we return pointer to the poller from commonclient.NewPoller instead of returning value?

Reply:
I don't see much of an advantage, maybe in a later PR.